### PR TITLE
Add handler for "getTokenById" API method.

### LINKED ISSUE #11: Rest endpoint - /api/v1/tokens/getTokenById
- Implement the endpoint with all filters
- Check the query time and add needed filters.
- Refer to https://mainnet-public.mirrornode.hedera.com/api/v1/docs/#/tokens/getTokenById

### PR DESCRIPTION
**Pull request by MarcKriguerAtHedera**
Signed-off-by: Marc Kriguer <marc.kriguer@swirldslabs.com>

Each of these 4 GET urls:

- http://localhost:8080/api/v1/tokens/107594
- http://localhost:8080/api/v1/tokens/107594?timestamp=1610640452612903002
- http://localhost:8080/api/v1/tokens/107594?timestamp=1610640421985772001
- http://localhost:8080/api/v1/tokens/107594?timestamp=lt:1610640452612903002
take somewhere between 65 and 80 ms each time.

### CODE REVIEW

File: src/main/java/com/swirlds/recordserver/TokensService.java [hunk 1]
```diff
@@ -143,4 +146,121 @@ private void getDefaultMessageHandler(ServerRequest request, ServerResponse resp
...
+        // no data retained for total_supply field
+        addIfNotNull(returnObject, "treasury_account_id", fields.getString("treasury", null));
+        addIfNotNull(returnObject, "type", fields.getString("tokenType", null));
+        addIfKeyNotNull(returnObject, "wipe_key", fields, "wipeKey");
+        // no data retained for custom_fees field
+
+        response.send(returnObject.build());
+    }
+
+    private void addIfNotNull(JsonObjectBuilder object, String name, String value) {
```
Comment (mgoelswirlds) [inline src/main/java/com/swirlds/recordserver/TokensService.java]: I actually like this and I think we should use this consistently in all our endpoints. Was wondering if we can move it to a utils class or something.

File: src/main/java/com/swirlds/recordserver/TokensService.java [hunk 2]
```diff
@@ -143,4 +146,121 @@ private void getDefaultMessageHandler(ServerRequest request, ServerResponse resp
...
+GET http://localhost:8080/api/v1/tokens/107594?timestamp=1610640452612903002
+### getTokenById specifiying an earlier timestamp
+GET http://localhost:8080/api/v1/tokens/107594?timestamp=1610640421985772001
+### getTokenById specifiying an earlier timestamp than the passed-in one
+GET http://localhost:8080/api/v1/tokens/107594?timestamp=lt:1610640421985772001
+**********************/
+    private void getTokenById(ServerRequest request, ServerResponse response) {
+        System.out.println("Calling into getTokenById!");
+	
+        final Optional<String> timestampQueryParam = request.queryParams().first("timestamp");
```
Comment (mgoelswirlds) [inline src/main/java/com/swirlds/recordserver/TokensService.java:164]: I am not sure If I am missing it, but I don't see filtering by tokenId

File: src/main/java/com/swirlds/recordserver/TokensService.java (hunk 1 above)
Comment (MarcKriguerAtHedera) [reply src/main/java/com/swirlds/recordserver/TokensService.java]: I am perfectly fine moving this; I was also thinking of maybe updating the `listTokens()` routine (same `TokenServices.java` file) to use it as well.  As far as existing utilities classes -- we have a "**utils**" directory with a `Utils.java` that has routines like toHex() and toBase64() that I think would be a good place to put these.  Or I could create a new utility class either in that directory, or just inside the `src/main/java/com/swirlds/recordserver` directory, if you prefer.

File: src/main/java/com/swirlds/recordserver/TokensService.java (hunk 2 above)
Comment (MarcKriguerAtHedera) [reply src/main/java/com/swirlds/recordserver/TokensService.java:164]: I know it had been there -- a call to request.path() used to set a whereClause -- but now I don't see it, either.  I'll recreate it and get back to you.

File: src/main/java/com/swirlds/recordserver/TokensService.java (hunk 1 above)
Comment (mgoelswirlds) [reply src/main/java/com/swirlds/recordserver/TokensService.java]: I think Utils is fine.

**Review by mgoelswirlds — APPROVED** [review]
LGTM

**Pull request merged**